### PR TITLE
[SPARK-37623][SQL] Support ANSI Aggregate Function: regr_intercept

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
`REGR_INTERCEPT` is an ANSI aggregate functions

**Syntax**: REGR_INTERCEPT(y, x)
**Arguments**: 
- **y**:The dependent variable. This must be an expression that can be evaluated to a numeric type.
- **x**:The independent variable. This must be an expression that can be evaluated to a numeric type.

**Examples**:
`select k, regr_intercept(v, v2) from aggr group by k;`

|  k |    regr_intercept(v, v2) |
|---|--------------------|
| 1  |          [NULL]            |
| 2 |     1.154734411       |

The algorithm refers https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance

The mainstream database supports `regr_intercept` show below:
**Teradata**
https://docs.teradata.com/r/kmuOwjp1zEYg98JsB8fu_A/MpkZBV~MSTZ~I84I~ezxNg
**Snowflake**
https://docs.snowflake.com/en/sql-reference/functions/regr_intercept.html
**Oracle**
https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/REGR_-Linear-Regression-Functions.html#GUID-A675B68F-2A88-4843-BE2C-FCDE9C65F9A9
**DB2**
https://www.ibm.com/docs/en/db2/11.5?topic=af-regression-functions-regr-avgx-regr-avgy-regr-count
**H2**
http://www.h2database.com/html/functions-aggregate.html#regr_intercept
**Postgresql**
https://www.postgresql.org/docs/8.4/functions-aggregate.html
**Sybase**
https://infocenter.sybase.com/help/index.jsp?topic=/com.sybase.help.sqlanywhere.12.0.0/dbreference/regr-intercept-function.html
**Presto**
https://prestodb.io/docs/current/functions/aggregate.html
**Exasol**
https://docs.exasol.com/sql_references/functions/alphabeticallistfunctions/regr_function.htm

### Why are the changes needed?
`REGR_INTERCEPT` is very useful.

### Does this PR introduce _any_ user-facing change?
'No'. New feature.

### How was this patch tested?
New tests.

### CODE REVIEW

**Comment by beliefer** [comment]
ping @MaxGekk cc @cloud-fan

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/linearRegression.scala [hunk 1]
```diff
@@ -291,3 +291,52 @@ case class RegrSlope(left: Expression, right: Expression) extends DeclarativeAgg
...
+      > SELECT _FUNC_(y, x) FROM VALUES (null, 1) AS tab(y, x);
+       NULL
+  """,
+  group = "agg_funcs",
+  since = "3.4.0")
+// scalastyle:on line.size.limit
+case class RegrIntercept(left: Expression, right: Expression) extends DeclarativeAggregate
+  with ImplicitCastInputTypes with BinaryLike[Expression] {
+
+  private val regrSlope = RegrSlope(left, right)
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/linearRegression.scala:313]: It looks tricky to have one more level of indirection, can we create `CovPopulation` and `VariancePop` here?

**Review by cloud-fan — APPROVED** [review]

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @beliefer and @cloud-fan for review.

**Comment by beliefer** [comment]
@MaxGekk @cloud-fan Thank you for review.

**Pull request closed without merging**